.werft/jobs/build/installer/installer.ts

```diff
@@ -0,0 +1,237 @@
+import { exec } from "../../../util/shell";
+import { getGlobalWerftInstance } from "../../../util/werft";
+import { getNodePoolIndex } from "../deploy-to-preview-environment";
+
+const blockNewUserConfigPath = './blockNewUsers';
+const workspaceSizeConfigPath = './workspaceSizing';
+const PROJECT_NAME = "gitpod-core-dev";
+const CONTAINER_REGISTRY_URL = `eu.gcr.io/${PROJECT_NAME}/build/`;
+const CONTAINERD_RUNTIME_DIR = "/var/lib/containerd/io.containerd.runtime.v2.task/k8s.io";
+
+export interface Installer {
+    configPath: string,
+    version: string,
+    proxySecretName: string
+    domain: string
+    previewName: string
+    imagePullSecretName: string
+    deploymentNamespace: string
+    analytics: Analytics,
+    withEELicense: boolean,
+    withVM: boolean,
+    workspaceFeatureFlags: string[],
+    gitpodDaemonsetPorts: GitpodDaemonsetPorts
+
+    init(slice: string): void
+    addPreviewConfiguration(slice: string): void
+    validateConfiguration(slice: string)
+    render(slice: string): void
+    postProcessing(slice: string): void
+    install(slice: string): void
+}
+
+export interface Analytics {
+    type: string,
+    token: string
+}
+
+export interface GitpodDaemonsetPorts {
+    registryFacade: number,
+    wsDaemon: number,
+}
+
+export function newInstaller(configPath: string, version: string, proxySecretName: string, domain: string, previewName: string, imagePullSecretName: string, deploymentNamespace: string, analytics: Analytics, withEELicense: boolean, withVM: boolean, workspaceFeatureFlags: string[], gitpodDaemonsetPorts: GitpodDaemonsetPorts): Installer {
+
+    const init = function init(slice) {
+        const werft = getGlobalWerftInstance()
```

Comment:
I would make `werft` a private field on the class that you pass in the constructor. That was you can use `this.werft` everywhere in the class.